src/preprocessing/passes/int_to_bv.cpp

```diff
@@ -147,30 +98,15 @@ Node intToBV(TNode n, NodeMap& cache)
   AlwaysAssert(size > 0);
   AlwaysAssert(!options::incrementalSolving());
 
-  vector<intToBV_stack_element> toVisit;
   NodeMap binaryCache;
   Node n_binary = intToBVMakeBinary(n, binaryCache);
-  toVisit.push_back(TNode(n_binary));
 
-  while (!toVisit.empty())
+  for (TNode current : NodeDfsIterable(n).inPostorder())
   {
-    // The current node we are processing
-    intToBV_stack_element& stackHead = toVisit.back();
-    TNode current = stackHead.d_node;
-
-    // If node is already in the cache we're done, pop from the stack
-    NodeMap::iterator find = cache.find(current);
-    if (find != cache.end())
-    {
-      toVisit.pop_back();
-      continue;
-    }
```

Comment:
same question for the similar place in the binarizer.

Reply:
Ah. I was wondering about this. I'll put it back in.

Reply:
Yoni and I talked about this offline and realized that if the traversal encounters an previously seen node, this optimization doesn't just avoid rebuilding it, it avoids traversing its DAG altogether.

The traversal iterator (right now) doesn't allow you to avoid traversing parts of the DAG. It does allow you to avoid rebuilding previously seen nodes, or skip the traversal altogether, if the whole DAG has been seen before, but that's the best it can do.

If this optimization is important, I should delay this PR and add a "skip traversal" feature to the iterator (or something like that) first. If this optimization isn't that important, I can approximate it as described above, and continue the PR.

My guess is that it is important, but I don't really know this part of the code very well. What do you think, @4tXJ7f?

Reply:
The pass should avoid traversing parts of the DAG that already have been processed. This is a general problem with the iterator and preprocessing passes.

Reply:
Now that the `skipIf` traversal modifier is on `master`, I've updated this PR. Why don't you take another look, Yoni and Mathias?